### ISSUE TITLE
How can I return an Observable instead a Promise?

### ISSUE THREAD

**Issue by JulioWar** [troyanskiy/rest-core#11]
Hello @troyanskiy 

I'm trying to migrate from **`Http` to `HttpCient`, I'm using `Angular 4.4.6`**. I'm trying to return a Observable instead a Promise and change the default property `asPromise` in the decorator `RestParams` but in the source code for this library I can see that you are only returning Promises in `IRestMethod`.

## This is my Code
```typescript
@Injectable()
@RestParams({
  url: BASE_API + '/providers',
  providersSubSet: 'providers',
  asPromise: false
})
export class ProviderResource extends Rest {
  @RestAction({
    isArray: false
  })
  query: IRestMethod<any, IProviderShort[]>;
}
``` 

What I need to do to change that behaviour ?

**Comment by troyanskiy** [comment]
Hello @JulioWar 
Indeed now the library works only with `Promises`.
If you really need to have `Observable` you can create it using [PromiseObservable](http://reactivex.io/rxjs/class/es6/observable/PromiseObservable.js~PromiseObservable.html)

**Comment by JulioWar** [comment]
Ok thanks @troyanskiy, I can work with that. Another question, Sorry to ask, but is possible to cancel or abort a request, I don't see any example about how can I do that. Thanks

**Comment by troyanskiy** [comment]
Yes, it's possible.
If you will set `toPromise: false` in the `ResourceAction` or `ResourceParams` you will have an [object](https://github.com/troyanskiy/ngx-resource-core/blob/master/src/Declarations.ts#L138-L142) as result. There you have `$abort` function to cancel the request.

**Comment by JulioWar** [comment]
Ok thanks @troyanskiy , I gonna try it

**Comment by httpete** [comment]
> Hello @JulioWar @troyanskiy 
> Indeed now the library works only with `Promises`.
> If you really need to have `Observable` you can create it using [PromiseObservable](http://reactivex.io/rxjs/class/es6/observable/PromiseObservable.js~PromiseObservable.html)

Can you show an example of how this is done thank you

**Comment by troyanskiy** [comment]
I'm going to check about observables.
Main issue is #49

**Comment by troyanskiy** [comment]
Hello @all
I did some refactoring and repository restructuring.
From now on new repository is located here https://github.com/troyanskiy/ngx-resource
Please create new issues in new repo.
I will close all the issues in that repo.

Also I have released 7.0.0-beta.5 version with `Observables` support. Please test it and create new issues in new repo if you will find some bugs.

To update to new version run
`npm i --save @ngx-resource/core@beta @ngx-resource/handler-ngx-http@beta`

### Breaking changes:
#### `ResourceGlobalConfig`
* removed `asPromise` and `asResourceResponse`
* added `returnAs` with enum type `ResourceActionReturnType`; Default is `ResourceActionReturnType.Observable`

#### `IResourceParams` and `IResourceAction`
* removed `asPromise` and `asResourceResponse`
* added `returnAs` with enum type `ResourceActionReturnType`; Default is `ResourceActionReturnType.Observable`

### Deprecation
* `IResourceMethod` -> `IResourceMethodPromise`
* `IResourceMethodStrict` -> `IResourceMethodPromiseStrict`
* `IResourceMethodFull` -> `IResourceMethodPromiseFull`
* `IResourceMethodStrictFull` -> `IResourceMethodPromiseStrictFull`
* `ResourceCRUD` -> `ResourceCRUDPromise`

### New
* `ResourceCRUDObservable` - same as ResourceCRUDPromise, but typed to Observable

I still have to update Docs and provide migration guide.

Thank you @all !!!
